### PR TITLE
feat(curriculum): CSS Box Model project

### PR DESCRIPTION
**Pull request by naomi-lgbt**
Checklist:

- [X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #XXXXX

Working on the box model project.

### CODE REVIEW

**Comment by nhcarrigan** [comment]
Going to hold off on writing tests until https://github.com/freeCodeCamp/freeCodeCamp/pull/42148 lands.

**Comment by nhcarrigan** [comment]
Realised that the boilerplates use the wrong `link` source. Will need to update.

File: curriculum/challenges/english/01-responsive-web-design/css-box-model/part-007.md
Comment (gikf) [inline curriculum/challenges/english/01-responsive-web-design/css-box-model/part-007.md:12]: As `href` attribute is expected as `./styles.css` it might be good to have that pointed in the challenge description, not only in one test.

File: curriculum/challenges/english/01-responsive-web-design/css-box-model/part-008.md
Comment (gikf) [inline curriculum/challenges/english/01-responsive-web-design/css-box-model/part-008.md:27]: Something is not right with how helper works, i.e. `getStyleDeclaration` is returning `undefined` seemingly always. This applies to other tests in this part and further challenges.

File: curriculum/challenges/english/01-responsive-web-design/css-box-model/part-008.md
Comment (gikf) [inline curriculum/challenges/english/01-responsive-web-design/css-box-model/part-008.md:41]:
```js
frame-runner.js:98 TypeError: Cannot read property 'getPropertyValue' of undefined
    at eval (eval at <anonymous> (VM33234 frame-runner.40c5018811bdc1204b53.js:2), <anonymous>:1:77)
    at HTMLDocument.<anonymous> (VM33234 frame-runner.40c5018811bdc1204b53.js:2)
    at u (VM33234 frame-runner.40c5018811bdc1204b53.js:2)
    at c (VM33234 frame-runner.40c5018811bdc1204b53.js:2)
```
This gets explicit error.

**Review by gikf — COMMENTED** [review]
Seems like CSS helper is not working here as expected, some more information on part 8.

File: curriculum/challenges/english/01-responsive-web-design/css-box-model/part-008.md
Comment (ShaunSHamilton) [reply curriculum/challenges/english/01-responsive-web-design/css-box-model/part-008.md:27]: This looks like the issue described here: https://github.com/freeCodeCamp/freeCodeCamp/pull/42266

**Comment by ShaunSHamilton** [comment]
@nhcarrigan Looks like you still need to rebase to get https://github.com/freeCodeCamp/freeCodeCamp/pull/42266

File: curriculum/challenges/english/01-responsive-web-design/css-box-model/part-008.md
Comment (gikf) [reply curriculum/challenges/english/01-responsive-web-design/css-box-model/part-008.md:27]: :+1: I can confirm this. After making change manually and restarting `npm run develop`, everything works.

File: curriculum/challenges/english/01-responsive-web-design/css-box-model/part-008.md
Comment (nhcarrigan) [reply curriculum/challenges/english/01-responsive-web-design/css-box-model/part-008.md:41]: That error should only throw when the previous test fails, where the camper hasn't added the `.canvas` selector yet.

File: curriculum/challenges/english/01-responsive-web-design/css-box-model/part-008.md
Comment (nhcarrigan) [reply curriculum/challenges/english/01-responsive-web-design/css-box-model/part-008.md:27]: Performed a rebase. Should be fixed.

File: curriculum/challenges/english/01-responsive-web-design/css-box-model/part-008.md
Comment (nhcarrigan) [reply curriculum/challenges/english/01-responsive-web-design/css-box-model/part-008.md:41]: Hmm - could maybe avoid the error with optional chaining. Will play with that.

Comment (ShaunSHamilton) [reply curriculum/challenges/english/01-responsive-web-design/css-box-model/part-008.md:41]: > avoid the error with optional chaining.

That is largely what I have been doing.

Comment (gikf) [reply curriculum/challenges/english/01-responsive-web-design/css-box-model/part-008.md:41]: Yeah, sorry, this was when I didn't have yet https://github.com/freeCodeCamp/freeCodeCamp/pull/42266 applied locally.

File: curriculum/challenges/english/01-responsive-web-design/css-box-model/part-001.md
Comment (gikf) [inline curriculum/challenges/english/01-responsive-web-design/css-box-model/part-001.md:44]: This confuses again the heck out of me, step can be passed just with adding `<!DOCTYPE html>`

Comment (moT01) [reply curriculum/challenges/english/01-responsive-web-design/css-box-model/part-001.md:44]: The reason is that the browser/preview automatically add those elements - and the tests are just checking that they're there.

File: curriculum/challenges/english/01-responsive-web-design/css-box-model/part-007.md
Comment (moT01) [inline curriculum/challenges/english/01-responsive-web-design/css-box-model/part-007.md:19]: **This lesson wasn't passing for me. I added the `link` element, but it's not getting added to the html...**

Screen Shot 2021-07-21 at 5 13 38 PM


**If I duplicate the element, it shows up:**

Screen Shot 2021-07-21 at 5 13 56 PM

😕 I tested in firefox and safari - rebased and tested again. All the same for me. I made some regex suggestions that should work for now - so we wouldn't need to rely on that getting rendered. I also added a test and reworded a few of them. Any idea what's going on here @nhcarrigan? I assume it was passing for you at one point. Maybe it's just something on my end.

```suggestion
assert(/<link/.test(code))
```

File: curriculum/challenges/english/01-responsive-web-design/css-box-model/part-007.md
Comment (moT01) [inline curriculum/challenges/english/01-responsive-web-design/css-box-model/part-007.md:25]:
````suggestion
Your `href` attribute should have the value `./styles.css`.

```js
assert(/href\s*=\s*('|")(\.\/)?styles\.css\1/.test(code));
````

File: curriculum/challenges/english/01-responsive-web-design/css-box-model/part-007.md
Comment (moT01) [inline curriculum/challenges/english/01-responsive-web-design/css-box-model/part-007.md:31]:
````suggestion
Your `rel` attribute should have the value `stylesheet`.

```js
assert(/rel\s*=\s*('|")\s*stylesheet\s*\1/.test(code));
```

Your `link` element should have `href="./styles.css"` and `rel="stylesheet"` attributes

```js
assert(/<\s*link(\s+href\s*=\s*("|')(\.\/)?styles\.css\2\s*rel=('|")\s*stylesheet\s*\4|\s+rel\s*=\s*('|")\s*stylesheet\s*\5\s*href\s*=\s*("|')(\.\/)?styles\.css\6)\s*\/?>/.test(code));
````

**Review by moT01 — COMMENTED** [review]
I had an issue with passing one of the lessons. Made some suggestions - not sure if anyone else wants to give it a try and see if they're having that issue. 

I was mostly just making sure the tests are all passing. But some other thoughts on this tutorial...
- I think it had a similar issue to the skyline I made. We didn't have the editable region when the projects were designed, so users would have had to find the correct spot to add code. It's not exclusively from those regions, but it feels a little too hand holdy. Basically every instruction tells you exactly what to add - and you know exactly where to add it. Not all that challenging. But you do get the repetition and practice.
- One minor thing was that the tutorial kind of abruptly ended. All of a sudden, I was done. Maybe a little heads up. "This is the last step" or "Only one more thing to add"
- The preview content is pretty much always cut off the screen and there's no horizontal scroll bar:
Screen Shot 2021-07-21 at 4 31 57 PM
I can resize the panes, but they get reset after each step. I can also zoom out or make the window really big. I guess that's not a huge deal, but something anyway.

File: curriculum/challenges/english/01-responsive-web-design/css-box-model/part-007.md
Comment (nhcarrigan) [reply curriculum/challenges/english/01-responsive-web-design/css-box-model/part-007.md:19]: Ah right, we made changes to the renderer to strip out these link elements.

**Review by moT01 — DISMISSED** [review]
I would be fine adding this - LGTM 🎉

File: curriculum/challenges/english/01-responsive-web-design/css-box-model/part-034.md
Comment (ShaunSHamilton) [inline curriculum/challenges/english/01-responsive-web-design/css-box-model/part-034.md:29]:
```suggestion
```

File: curriculum/challenges/english/01-responsive-web-design/css-box-model/part-036.md
Comment (ShaunSHamilton) [inline curriculum/challenges/english/01-responsive-web-design/css-box-model/part-036.md:41]:
```suggestion
```

File: curriculum/challenges/english/01-responsive-web-design/css-box-model/part-046.md
Comment (ShaunSHamilton) [inline curriculum/challenges/english/01-responsive-web-design/css-box-model/part-046.md:10]: I would more naturally think _'counter-clockwise by 0.2deg'_ . Thoughts?

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
This project is _ok_. However, I do not feel like anything is accomplished - I do not know what a Rothko is, and did not feel like I was done with it.

Sorry, if that sounds blunt 😄

**Comment by ShaunSHamilton** [comment]
> I think we should just keep trying to improve as we build more.

This is definitely the case with...well... all the curriculum development. I am happy with some not being as exciting to do, provided they give a decent solidification of concepts (repetition and such). So, if you are happy with this, I am too (after removing the `console.log`s).

**Comment by nhcarrigan** [comment]
I can sweep back through and try to improve the instructions too

**Review by ShaunSHamilton — DISMISSED** [review]
I am happy with this. Those changes do make the project feel more resolved 👍 

Nice job, Nich 💯

**Comment by ieahleen** [comment]
What do I need to do to see the project other than changing the `.env` file to have `SHOW_UPCOMING_CHANGES=true`?

**Comment by ojeytonwilliams** [comment]
@ieahleen that should be enough.  If it fails, `npm ci && npm run clean-and-develop` should be enough, but it takes a few minutes to complete.

**Comment by moT01** [comment]
I fixed the conflicts here - this should be good to go. I will approve once tests pass 👍

File: curriculum/challenges/_meta/css-box-model/meta.json
Comment (moT01) [inline curriculum/challenges/_meta/css-box-model/meta.json:5]: I went through the first couple lessons on this again after fixing those conflicts. After completing part one on this, I get sent to part one on the Picasso painting, then to part 2 here, and then to part 2 on the painting, and so on... Pretty sure it's cause they share this order number. I didn't verify it, but they are both using 11 - seems likely that it's the culprit.

**Review by moT01 — CHANGES_REQUESTED** [review]
Sorry we didn't get this merged when it was ready @nhcarrigan - but there's some more issues now 😞 - I can maybe give it a fix tomorrow or something if you want to be done with it.

File: curriculum/challenges/_meta/css-box-model/meta.json
Comment (nhcarrigan) [reply curriculum/challenges/_meta/css-box-model/meta.json:5]: Yep. That's the issue - ran in to it with the piano project today.
Will fix in the morning 🙂

**Review by moT01 — APPROVED** [review]
LGTM 🎉 

I noticed that this shares the meta order with the piano project - it will have to be changed over there if this gets merged first.

**Review by ShaunSHamilton — APPROVED** [review]
Still LGTM 👍 

Nich, might want to rebase the Piano PR, for us to get a sense of how it will be when we merge both.

**Pull request merged**
